openfisca_core/holders.py

```diff
@@ -251,6 +270,71 @@ def graph(self, edges, get_input_variables_and_parameters, nodes, visited):
             return
         formula.graph_parameters(edges, get_input_variables_and_parameters, nodes, visited)
 
+    def get_memory_usage(self):
+        """
+            Gets data about the virtual memory usage of the holder.
+
+            :returns: Memory usage data
+            :rtype: dict
+
+            Exemple:
+
+            >>> holder.get_memory_usage()
+            >>> {
+            >>>    'nb_arrays': 12,  # The holder contains the variable values for 12 different periods
+            >>>    'nb_cells_by_array': 100, # There are 100 entities (e.g. persons) in our simulation
+            >>>    'cell_size': 8,  # Each value takes 8B of memory
+            >>>    'dtype': dtype('float64')  # Each value is a float 64
+            >>>    'total_nb_bytes': 10400  # The holder uses 10.4kB of virtual memory
+            >>>    }
+        """
+
+        usage = dict(
+            nb_cells_by_array = self.entity.count,
+            dtype = self.variable.dtype,
+            )
+
+        if self._array is not None:
+            # Only used when definition period is ETERNITY
+            usage.update(dict(
+                nb_arrays = 1,
+                total_nb_bytes = self._array.nbytes,
+                cell_size = self._array.itemsize,
+                ))
+            return usage
+        elif self._array_by_period is not None:
+            nb_arrays = sum([
+                len(array_or_dict) if isinstance(array_or_dict, dict) else 1
+                for array_or_dict in self._array_by_period.itervalues()
+                ])
+            array = self._array_by_period.itervalues().next()
```

Comment:
Why not just take the first item ?

Reply:
There is no order in a dictionary, so no first element either.
The alternative would be 

```py
first_key = self._array_by_period.keys()[0]
first_value = self._array_by_period[first_key]
```

which is more heavy

Reply:
You were right, I forgot about `self._array_by_period.values()[0]`, which may be simpler.